### PR TITLE
[INTEL MKL] Add MKL Conv + Bias + LeakyRelu Fusion

### PR DESCRIPTION
**Pull request by retonym**
Add Conv + Bias + LeakyRelu fusion MKL implementation

### CODE REVIEW

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 1]
```diff
@@ -1588,6 +1589,7 @@ REGISTER_TEST_ALL_TYPES(NodeRewrite_FusedConv2D_Positive1);
               "i:1, i:1} } }"                                                 \
               " attr { key: 'fused_ops'        value { list: {s: 'Relu'} } }" \
               " attr { key: 'epsilon'          value { f: 0.001 }}"           \
+              " attr { key: 'leakyrelu_alpha'  value { f: 0.2 }}"                    \
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass_test.cc:1592]: Nit: would you mind moving the `\` at the end to align with the rest of the lines? Thank you!

File: tensorflow/core/grappler/optimizers/mkl_remapper_test.cc [hunk 2]
```diff
@@ -73,27 +75,51 @@ class MklRemapperTest : public GrapplerTest {
...
+        ops::Identity(fetch, ops::Relu6(activate, add));
+      } else if (activation == "Elu") {
+        ops::Identity(fetch, ops::Elu(activate, add));
+      } else if (activation == "LeakyRelu") {
+        ops::Identity(fetch, ops::internal::LeakyRelu(activate, add));
       } else {
-        ops::Identity(s.WithOpName("fetch"), add);
+        DCHECK(activation == "None");
+        ops::Identity(fetch, add);
       }
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/mkl_remapper_test.cc:107]: The logic is the same in all three cases (`AddN, AddV2, Add`). Can we refactor this to a function (or a lambda function)?

File: tensorflow/core/grappler/optimizers/remapper.cc [hunk 3]
```diff
@@ -361,12 +361,7 @@ bool IsDeviceCompatible(const RemapperContext& ctx, Pattern& matched) {
 }
 
 bool IsSupportedActivation(const NodeDef& node) {
-// Disable LeakyRelu temporarily before MKL PR is merged.
-#ifndef INTEL_MKL
   return IsRelu(node) || IsRelu6(node) || IsElu(node) || IsLeakyRelu(node);
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/remapper.cc:364]: (No action required until #42173 is merged.)
This (and probably some other parts of this PR) will have merge conflict with #42173. Sorry I couldn't get to #42173 sooner!

File: tensorflow/core/grappler/optimizers/remapper_test.cc [hunk 4]
```diff
@@ -984,7 +974,7 @@ TEST_F(RemapperTest, FuseConv2DWithSqueezeAndBias) {
   ASSERT_EQ(tensors.size(), 1);
   test::ExpectTensorNear<float>(tensors[0], tensors_expected[0], 1e-6);
 }
-#endif
+#endif  // !INTEL_MKL
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/remapper_test.cc:977]: So this means `FuseConv2DWithBatchNorm`, `FuseConv2DWithBatchNormAndActivation`, and `FuseConv2DWithSqueezeAndBias` tests will remain disabled for MKL build, right?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have a few comments.

Note to self: Don't pull this in until #42173 is merged.

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc (hunk 1 above)
Comment (retonym) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc]: The `\` are aligned.

File: tensorflow/core/grappler/optimizers/mkl_remapper_test.cc (hunk 2 above)
Comment (retonym) [reply tensorflow/core/grappler/optimizers/mkl_remapper_test.cc]: Thanks for the advice. I rewrite this part of codes.

File: tensorflow/core/grappler/optimizers/remapper.cc (hunk 3 above)
Comment (retonym) [reply tensorflow/core/grappler/optimizers/remapper.cc:364]: Yeah, I will fix the conflicts after https://github.com/tensorflow/tensorflow/pull/42173 is merged

File: tensorflow/core/grappler/optimizers/remapper_test.cc (hunk 4 above)
Comment (retonym) [reply tensorflow/core/grappler/optimizers/remapper_test.cc:977]: Yes, these fusions are not supported by MKL currently.

**Review by penpornk — APPROVED** [review]
Thank you for the quick fixes!

**Comment by penpornk** [comment]
#42173 is being merged (will take a while). I'll add the `ready-to-pull` tag back to this PR after #42173 is done merging.

**Comment by penpornk** [comment]
https://github.com/tensorflow/tensorflow/pull/42173 is merged. Adding `ready-to-pull` tag.

**Review by penpornk — CHANGES_REQUESTED** [review]
https://github.com/tensorflow/tensorflow/pull/42173 is merged. Could you please resolve the conflicts? Thank you!

**Comment by retonym** [comment]
The conflicts are fixed.

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Review by gbaned — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Pull request merged**